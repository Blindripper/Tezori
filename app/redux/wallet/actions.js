import {
  SET_WALLET,
  SET_IS_LOADING,
  SET_WALLET_FILENAME,
  SET_WALLET_LOCATION,
  SET_PASSWORD,
  SET_IDENTITIES,
  ADD_NEW_IDENTITY,
  UPDATE_IDENTITY,
  ADD_NEW_ACCOUNT
  LOGOUT,
} from './types';
import actionCreator from '../../utils/reduxHelpers';


export const setWallet = actionCreator(SET_WALLET, 'wallet');
export const setIsLoading = actionCreator(SET_IS_LOADING, 'isLoading');
export const setWalletFileName = actionCreator( SET_WALLET_FILENAME, 'walletFileName' );
export const updateWalletLocation = actionCreator( SET_WALLET_LOCATION, 'walletLocation' );
export const setPassword = actionCreator(SET_PASSWORD, 'password');
export const setIdentities = actionCreator(SET_IDENTITIES, 'identities');
export const addNewIdentity = actionCreator(ADD_NEW_IDENTITY, 'identity');
export const updateIdentity = actionCreator(UPDATE_IDENTITY, 'identity');
export const addNewAccount = actionCreator( ADD_NEW_ACCOUNT, 'publicKeyHash', 'account' );
export const logout = actionCreator(LOGOUT);
