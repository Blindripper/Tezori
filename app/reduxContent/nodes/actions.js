import {
  SET_SELECTED,
  ADD_NODE,
  REMOVE_NODE,
  UPDATE_NODE,
  CLEAR_STATE
} from './types';
import actionCreator from '../../utils/reduxHelpers';

export const setSelected = actionCreator(SET_SELECTED, 'selected', 'target');
export const addNode = actionCreator(ADD_NODE, 'node');
export const removeNode = actionCreator(REMOVE_NODE, 'name');
export const updateNode = actionCreator(UPDATE_NODE, 'node');
export const clearState = actionCreator(CLEAR_STATE);
