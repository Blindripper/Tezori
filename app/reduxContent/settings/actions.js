import {
  SET_SELECTED,
  SET_LOCAL,
  ADD_NODE,
  REMOVE_NODE,
  UPDATE_NODE,
  CLEAR_STATE
} from './types';
import actionCreator from '../../utils/reduxHelpers';

export const setSelected = actionCreator(SET_SELECTED, 'selected', 'target');
export const setLocal = actionCreator(SET_LOCAL, 'local');
export const addNode = actionCreator(ADD_NODE, 'node');
export const removeNode = actionCreator(REMOVE_NODE, 'name');
export const updateNode = actionCreator(UPDATE_NODE, 'node');
export const clearState = actionCreator(CLEAR_STATE);
