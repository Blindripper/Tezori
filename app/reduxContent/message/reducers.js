import { fromJS } from 'immutable';
import { LOGOUT } from '../wallet/types';
import { CLEAR_MESSAGE_STATE, ADD_MESSAGE } from './types';

const emptyMessage = fromJS({
  message: '',
  isError: false,
  hash: '',
  localeParam: 0
});

const initState = fromJS({
  message: {}
});

export default function messages(state = initState, action) {
  switch (action.type) {
    case CLEAR_MESSAGE_STATE:
      return initState;
    case ADD_MESSAGE: {
      const message = emptyMessage
        .set('message', action.message)
        .set('isError', action.isError)
        .set('hash', action.hash)
        .set('localeParam', action.localeParam);
      return state.set('message', message);
    }
    case LOGOUT:
      return initState;
    default:
      return state;
  }
}
