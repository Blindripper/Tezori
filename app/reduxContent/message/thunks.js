import { createMessage } from './actions';

export function addMessage(message, isError, hash = '') {
  return dispatch => {
    dispatch(createMessage(message, isError, hash));
  };
}
