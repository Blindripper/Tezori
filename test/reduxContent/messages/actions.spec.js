import * as actions from '../../../app/reduxContent/message/actions';
import configureStore from 'redux-mock-store';

const mockStore = configureStore();
const store = mockStore();

beforeEach(() => { 
    store.clearActions();
});

describe('Messages clear and create', () => {
    it('should return exact type', () => {
        store.dispatch(actions.clearMessageState());
        expect(store.getActions()).toMatchSnapshot();
    });

    it('should return exact type and payload', () => {
        store.dispatch(actions.createMessage('test',true));
        expect(store.getActions()).toMatchSnapshot();
    });
})