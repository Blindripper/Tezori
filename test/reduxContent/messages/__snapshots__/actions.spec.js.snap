// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Messages clear and create should return exact type 1`] = `
Array [
  Object {
    "type": "CLEAR_MESSAGE_STATE",
  },
]
`;

exports[`Messages clear and create should return exact type and payload 1`] = `
Array [
  Object {
    "isError": true,
    "message": "test",
    "type": "ADD_MESSAGE",
  },
]
`;
